### ISSUE TITLE
Component renders twice for each iteration with `for`

### ISSUE THREAD

**Issue by simonrepp** [slint-ui/slint#11047]
Hello Slint team!

### Bug Description

- I'm using `for` to repeatedly render a relatively simple custom component with input properties

  ```slint
    for list_item[index] in list_items : ListItem {
        ListItem {
            active: index == list_items.length - 1;
            index: index;
            list_item: list_item;
        }
    }
   ```

- Although I'm only providing/iterating an array of **2** structs, the component renders **4** times

   ```slint
   List {
       list_items: [
           { value: "Red", variant: Variant.Red },
           { value: "Blue", variant: Variant.Blue }
       ];
   }
   ```

- Also in debugging the `init` callback for the component is triggered 4 times (twice for each `for` iteration).

   Output on the console (`init => debug(active, list_item);`):
   ```
   false { value: , variant: Blue }           <-- (probably faulty) iteration/instance with default values
   false { value: Red, variant: Red }
   false { value: , variant: Blue }           <-- (probably faulty) iteration/instance with default values
   true { value: Blue, variant: Blue }
   ```

To give you a visual idea of the problem, a screenshot: 

<img width="648" height="516" alt="Image" src="https://github.com/user-attachments/assets/bdd87b8e-a4ce-4fef-a623-645606ded537" />

---

I've provided a minimal reproduction example project repository here: https://github.com/simonrepp/slint_repro_repetition_opacity/tree/main

P.s.: I have already heavily edited the original bug report as I was able to track the problem further down (to its current form).

### Reproducible Code (if applicable)

```slint
enum Variant {
    Blue,
    Red
}

struct ListItemData {
    value: string,
    variant: Variant
}

component ListItem inherits HorizontalLayout {
    in property <bool> active;
    in property <int> index;
    in property <ListItemData> list_item;

    Rectangle {
        background: list_item.variant == Variant.Blue ? blue : red;

        // This prints 4 times in the console, although only 2 list items
        // should go through initialization
        init => debug(active, list_item);

        HorizontalLayout {
            alignment: center;
            VerticalLayout {
                alignment: center;
                spacing: 1rem;

                Text {
                    text: index;
                }

                Text {
                    text: list_item.variant == Variant.Blue ? "Blue" : "Red";
                }

                Text {
                    text: list_item.value;
                }
            }
        }
    }
}

component List inherits HorizontalLayout {
    in property <[ListItemData]> list_items;

    // We iterate the list items and render each as a component,
    // differentiating the last one through the `active` property
    for list_item[index] in list_items : ListItem {
        ListItem {
            active: index == list_items.length - 1;
            index: index;
            list_item: list_item;
        }
    }
}

export component ListExample inherits HorizontalLayout {
    // A list with two items: red/blue
    List {
        list_items: [
            { value: "Red", variant: Variant.Red },
            { value: "Blue", variant: Variant.Blue }
        ];
    }
}
```

### Environment Details

- Slint Version: 1.15.1
- Platform/OS: Debian 13, KDE 6.3.6, Wayland
- Programming Language: Rust (but the issue can be reproduced with pure slint)
- Backend/Renderer: Qt (6.8.2)

### Product Impact

As the reported issue seems to be a fundamental glitch in one of slint's core mechanics (repetition of components), I suppose many projects will potentially benefit from its resolution.

I personally am using Slint to build a graphical user interface for *Faircamp* (a static site generator for audio – https://simonrepp.com/faircamp/). Later this year I might also port a graphical user interface for a static site generator for video from Tauri to slint (*Hyper 8 Video System* – https://simonrepp.com/hyper8/).

### Thanks

Thank you for your time and for your work on slint, I'm very impressed with what you built! Keep it up!
Simon

**Comment by simonrepp** [comment]
Digging yet deeper I eventually realized that I missed the forest for the trees: I had somehow accidentally nested the affected component within itself in the `for` construct, which explains all of the weird stuff I observed:

```slint
    for list_item[index] in list_items : ListItem {
        ListItem {
```

So, entirely my bad after all. Hope I could catch and close this before any time was wasted on triaging, otherwise my sincere apologies. Thank you again for slint, fantastic work!
